src/transaction.js

```diff
@@ -434,9 +434,11 @@ const transaction = {
   calculateTxWeight(txData) {
     let txSize = this.txToBytes(txData).length;
 
-    // XXX Parents are calculated only in the server but we need to consider them here
-    // Parents are always two and have 32 bytes each
-    txSize += 64
+    // If parents are not in txData, we need to consider them here
+    if (!txData.parents || !txData.parents.length || txData.parents.length === 0) {
+      // Parents are always two and have 32 bytes each
+      txSize += 64;
```

Comment:
I think we should raise an exception if `txData.parents.length > 2`, and use `txSize += 64 - 32 * txData.parents.length`.

Reply:
Done.

Also added a test case for that error.

Should we also throw an error when `parents.length == 1`?

Reply:
Great! Initially I thought about it but maybe the application will fill the other parent later. As more than two parents is always invalid, I feel it is enough to check this condition. What do you think?

Reply:
I agree

A possible use case for single parent validation would be someone wanting to "chain" transactions using a single parent, leaving the other one for automatic selecion

I think this is ready to merge, what do you think?

Reply:
Yes, it is.